### PR TITLE
FOGL-2879 Modbus restructing and caching of large register set reads

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: modbus_cache.cpp [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+/**
+ * Destructor for register range
+ */
+ModbusCacheManager::SlaveCache::RegisterRanges::~RegisterRanges()
+{
+	m_ranges.clear();
+}
+
+/**
+ * Add a register to the range of resisters.
```
Comment (pintomax) [inline modbus_cache.cpp]: Typo: resisters

File: modbus_cache.cpp [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+			for (map<int,int>::iterator it2 = m_ranges.begin(); it2 != m_ranges.end(); it2++)
+			{
+				if (it2->first == it1->first && it2->second == it1->second)
+				{
+					continue;	// it1 and it2 point to same range
+				}
+				if (it1->second + 1 == it2->first)
+				{
+					it1->second = it2->second;
+		Logger::getLogger()->info("Combined range at end %d, %d", it1->first, it1->second);
```
Comment (pintomax) [inline modbus_cache.cpp]: Fix indentation

File: include/modbus_south.h [hunk 3]
```diff
@@ -101,5 +195,141 @@ class Modbus {
...
+ * Under each source there is a set od ranges, each range represents a contiguous range
+ * of registers that are used in the modbus map. Only ranges that are cgreate than the
+ * defined threshold length are actually cached.
+ *
+ * During the reading stage of the map, the cache manager is called to register items,
+ * where an item is a slave, source and register num,ber tripple. The object structure
+ * is built to represent those ranges.
+ *
+ * Once reading of the modbus mao is complete the createCaches method of the manager is
+ * called to create the actual caches themselves. This will recurse down t the register
```
Comment (pintomax) [inline include/modbus_south.h]: Typo: down t the

**Review by pintomax — DISMISSED** [review]

**Comment by MarkRiddoch** [comment]
@ashwinscale Are you able to test this against the GA setup?

**Review by pintomax — DISMISSED** [review]

**Review by sstefano69 — APPROVED** [review]

**Pull request merged**